src/main/scala/firrtl/Utils.scala

```diff
@@ -691,10 +701,21 @@ object Utils {
      def flip(): Field = Field(field.name, field.flip.flip, field.tpe)
 
      def getType(): Type = field.tpe
-     def toPort(): Port = Port(NoInfo, field.name, field.flip.toDirection, field.tpe)
+     def toPort(info: Info = NoInfo): Port =
+       Port(info, field.name, field.flip.toDirection, field.tpe)
    }
 
    implicit class TypeUtils(t: Type) {
+     // TODO Should these be more safe? (ie. error on UnknownType)
+     def isGround: Boolean = t match {
+       case (_: UIntType | _: SIntType | _: ClockType) => true
+       case _ => false
+     }
+     def isAggregate: Boolean = t match {
+       case (_: BundleType | _: VectorType) => true
+       case _ => false
+     }
+
```

Comment:
Yeah I thought about doing it that way, I originally split it because I wasn't sure if ClockType was ground, but it actually is so I think you're right. 

Do you think it would make sense to still have isAggregate as `def isAggregate = !t.isGround`?